internal/redispool/naive.go

```diff
@@ -206,6 +206,26 @@ func (kv *naiveKeyValue) HSet(key, field string, fieldValue any) error {
 	}).err
 }
 
+func (kv *naiveKeyValue) HDel(key, field string) Value {
+	var reply any
+	err := kv.maybeUpdateValues(redisGroupHash, key, func(li []any) ([]any, updaterOp, error) {
+		idx, ok, err := hsetValueIndex(li, field)
+		if err != nil {
+			return li, readOnly, err
+		}
+		if !ok {
+			return li, readOnly, redis.ErrNil
+		}
+
+		reply = li[idx]
+		return li, readOnly, nil
+	}).err
+	if err != nil {
+		return Value{reply: 0}
+	}
+	return Value{reply: reply}
```

Comment:
yeah this implementation needs to actually remove the value from li + return the removed count. I didn't test this, but here is my attempt at fixing in a github comment box.

```suggestion
	var removed int64
	err := kv.maybeUpdateValues(redisGroupHash, key, func(li []any) ([]any, updaterOp, error) {
		idx, ok, err := hsetValueIndex(li, field)
		if err != nil || !ok {
			return li, readOnly, err
		}
		
		// Value exists at idx so remove it from the list
		removed = 1
		li = append(li[:idx], li[idx+1]...)
		return li, write, nil
	}).err
	return Value{reply: removed, err: err}
```

Reply:
Ahh. This makes sense. Thanks for the explanation 😄